### DISCUSSION TITLE
got wallpaper engine kde plugin work on steam deck

### DISCUSSION THREAD

**Discussion by Willlost** [catsout/wallpaper-engine-kde-plugin#285] [Show and tell]
Its just about installing lots of dependencies, really. Took me 6 hours to figure everything out. 

https://user-images.githubusercontent.com/39188259/226128028-3c157f09-2123-448e-86bc-c596d5a01008.mp4

**Comment by Willlost** [comment]
Thanks to catsout for this incredible plugin!

**Reply by UltraHydro** [reply]
Could you help me lol

**Reply by krgrabowski** [reply]
nice but could you tell us which ones you've installed?

**Reply by Willlost** [reply]
@krgrabowski you can use all the wallpapers supported by this plugin. I think cats out have them listed in readme. Personally I only installed this one and a bilibili interactive wallpaper, which works as well as on windows.

**Comment by Willlost** [comment]
To install this plugin on steam deck you will need to install lots of missing packages (according to the error messages) and edit the cmake cache file in the build folder (provide paths for cmake to locate the required packages). Good luck.

**Reply by krgrabowski** [reply]
Thanks a lot i've asked, cause i thought it might help me with my vanilla arch system, since steamos is an arch based system.
Thing is i use kde plasma 6 and valve decided to stay on kde 5, most likely to not to break anything.
The installation failed and i thought maybe i'm still missing some dependency, but no i just had to use the AUR package with the makepkg package builder.
Also i had to delete the plugin entry in the usr/share/metainfo, otherwise pacman refuses to install the pkg, probbably a good idea to mention it in the readme.
Unfortunatly KDE 6 and wayland support are kind of still experimental and wip, so many wallpapers do not work, especially scene wallpapers do not.

**Comment by BrickCraftDream** [comment]
It took me like 10 hours to figure out, what I had to do is go into the CMakeCache.txt file and replace the line that starts with `X11_X11_INCLUDE_PATH:PATH=` with `X11_X11_INCLUDE_PATH:PATH=usr/include/x11` and to not clone the qt6 repo. For the make part to work I had to, I think, install some dependencies and run this command `sudo ln -s /home/deck/miniconda3/pkgs/sysroot_linux-64-2.12-he073ed8_17/x86_64-conda-linux-gnu/sysroot/usr/lib64/libpthread_nonshared.a /usr/lib64/libpthread_nonshared.a`. Also I had to limit the make command (`make -j4`) because my Steam Deck kept crashing. Keep in mind that not every scene or web works, nearly every 3d scene I tested doesn't work

**Reply by BrickCraftDream** [reply]
[Here](https://drive.google.com/file/d/1iQehUvvk0GJxepB2YTb5hzUJKek3n0kh/view?usp=sharing) is the content of `/usr/lib/qt/qml/com/github/catsout/wallpaperEngineKde/` for anyone who might need it (it's the lib)
